app/src/androidTest/java/com/alexzh/testapp/ui/home/HomeActivityTest.kt

```diff
@@ -18,77 +18,95 @@ import org.junit.runner.RunWith
 @RunWith(AndroidJUnit4::class)
 class HomeActivityTest {
 
-    @Rule
-    @JvmField
+    /**
+     * The {@link ActivityTestRule} provides functional testing of a single {@link Activity}.
+     * The Espresso framework requires it for launching Activity for testings.
+     *
+     * We can configure activity before running in the following ways:
+     *  - Using the {@link #ActivityTestRule(Class, boolean, boolean)} where the second parameter
+     *  is initialTouchMode and the third one is launchActivity. If launch activity is true the
+     *  Activity will be launched before the test case. It means we cannot configure it before
+     *  executing the test case. If we want to configure which intent launch activity and pass
+     *  additional parameters we can execute {@see activity#launchActivity(Intent)} method with
+     *  required intent.
+     *  - Using the {@link ActivityTestRule#beforeActivityLaunched) method we can do additional
+     *  action before executing test case, like configure Mock objects, clean database, etc.)
+     *
+     * Note:
+     *  - a field created for JUnit Rule should be public, not static, and a subtype of TestRule
+     *  or MethodRule (https://junit.org/junit4/javadoc/4.12/org/junit/Rule.html)
+     *  - In Kotlin we can use @get: Rule or @Rule @JvmField annotations for JUnit Rule fields
+     */
+    @get: Rule
     val activity = ActivityTestRule(HomeActivity::class.java)
 
     /**
      * Test case:
-     *  - Toolbar should have a text "TestApp"
+     *  - HomeActivity should have options menu
+     *  - SettingsActivity should be opened after pressing on "Settings" in options menu
      *
      * Notes:
-     *  - the "Toolbar" view has ID: toolbar (R.id.toolbar)
-     *  - the "TestApp" string can be found in strings.xml file as app_name const (R.string.app_name)
+     *  - the "Settings" text can be found in strings.xml file as settings_action const
+     *  (R.string.settings_action)
      *
      * Hint(s):
-     *  - the toolbar title can be checked with ToolbarMatcher#withToolbarTitle ViewMatcher
-     *  (can be not implemented yet).
+     *  - Intent.intended(IntentMatcher) allows us to verify that Activity was opened.
+     *  The Intents component should be initialized before with the Intents#init() method
```

Comment:
before what? I understand that you imply the test but it reads strange. So `before using`, `before the test`, `before running the test`, `before executing the test` etc

Reply:
fixed